### PR TITLE
refactor(database): use SupersetResultSet on SqlaTable.get_df()

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Currently `SqlaTable.get_df()` is fetching data using `cursor.fetchall()`  instead of `BaseEngineSpec.fetch_data()`, which contains critical customizations for multiple databases. In addition, instead of constructing the `DataFrame` using `pd.DataFrame.from_records()`, we can leverage `SupersetResultSet` which is already implemented on SQL Lab, and which behind the scenes is using `PyArrow` for storing the datastructure. This is a yet another step in the direction of using `SupersetResultSet`, and therefore `PyArrow`,  instead of Pandas DataFrames whenever possible.

### TEST PLAN
CI

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Ping: @robdiciuccio

### CODE REVIEW

File: superset/typing.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
     str, str, Optional[str], Optional[str], Optional[int], Optional[int], bool
 ]
 DbapiDescription = Union[List[DbapiDescriptionRow], Tuple[DbapiDescriptionRow, ...]]
-DbapiResult = List[Union[List[Any], Tuple[Any, ...]]]
+DbapiResult = Sequence[Union[List[Any], Tuple[Any, ...]]]
```
Comment (villebro) [inline superset/typing.py:28]: for some reason `MyPy` was giving me errors on this locally (I'm probably running a newer version than CI). As I know some Dbapis are actually returning tuples, I think this makes sense anyway.

**Review by dpgaspar — APPROVED** [review]
LGTM

File: superset-frontend/cypress-base/cypress/integration/explore/visualizations/table.test.ts [hunk 2]
```diff
@@ -119,7 +129,7 @@ describe('Visualization > Table', () => {
   it('Test table with percent metrics and groupby', () => {
     const formData = {
       ...VIZ_DEFAULTS,
-      percent_metrics: NUM_METRIC,
+      percent_metrics: PERCENT_METRIC,
```
Comment (villebro) [inline superset-frontend/cypress-base/cypress/integration/explore/visualizations/table.test.ts:132]: For some weird reason this test was flaking out when run with `NUM_METRIC` (I replicated this test locally, but was not able to make it fail). Switching to `PERCENT_METRIC` below seems to have solved it.

**Review by robdiciuccio — APPROVED** [review]
LGTM!

**Pull request merged**